File: docs/versioned_docs/version-1.10.0/API-Reference/python-examples/api-openai-responses/additional-configuration-for-openai-client-libraries.py
import os

from openai import OpenAI

base = (os.environ.get("LANGFLOW_URL") or os.environ.get("LANGFLOW_SERVER_URL", "")).rstrip("/")
api_key = os.environ.get("LANGFLOW_API_KEY", "")
flow_id = os.environ.get("FLOW_ID", "")

client = OpenAI(
    base_url=f"{base}/api/v1/",
    default_headers={"x-api-key": api_key},
    api_key="dummy-api-key",  # Required by OpenAI SDK but not used by Langflow
)

try:
    response = client.responses.create(
        model=flow_id,
        input="There is an event that happens on the second wednesday of every month. What are the event dates in 2026?",
    )
except Exception as exc:
    # Empty bootstrap flows return an error body; use a flow with ChatInput + ChatOutput in the UI.
    print(exc)
else:
    try:
        print(response.output_text)
    except Exception:
        print(response)

File: docs/versioned_docs/version-1.12.0/API-Reference/javascript-examples/api-openai-responses/additional-configuration-for-openai-client-libraries.ts
import OpenAI from "openai";

const client = new OpenAI({
  baseURL: "LANGFLOW_SERVER_URL/api/v1/",
  defaultHeaders: {
    "x-api-key": "LANGFLOW_API_KEY"
  },
  apiKey: "dummy-api-key" // Required by OpenAI SDK but not used by Langflow
});

const response = await client.responses.create({
  model: "FLOW_ID",
  input: "There is an event that happens on the second wednesday of every month. What are the event dates in 2026?"
});

console.log(response.output_text);
